### ISSUE TITLE
pallet-session doesn't enable setting weights, removing the ability to weight Grandpa

### ISSUE THREAD

**Issue by kayabaNerve** [paritytech/polkadot-sdk#2349]
pallet-session doesn't expose an API to set weights nor an API to retrieve weights from some other pallet.

Then, `OneSessionHandler` takes a list of validators yet doesn't take the weights associated with them.

[Babe's implementation of `OneSessionHandler` sets the weight to 1.](https://paritytech.github.io/polkadot-sdk/master/src/pallet_babe/lib.rs.html#967)

[Grandpa's implementation of `OneSessionHandler` also sets the weight to 1.](https://paritytech.github.io/polkadot-sdk/master/src/pallet_grandpa/lib.rs#585)

Due to Grandpa's `OneSessionHandler`'s usage of private members (notably `SetIdSession`), it's not possible to properly use Grandpa without pallet-session, making it impossible to use pallet-grandpa with weights.

**Comment by kayabaNerve** [comment]
Correct, NOT execution weights (related to fees/the amount of transactions a block will contain).

**Comment by bkchr** [comment]
Can you may provide more information on how you want to weight the authorities? And if you don't need `pallet_session`, can you not just use [`schedule_change`](https://github.com/paritytech/polkadot-sdk/blob/50390950d87fb745ec5ff46be698cac93113efb7/substrate/frame/grandpa/src/lib.rs#L478) in your implementation?

**Comment by kayabaNerve** [comment]
By units of stake on a scale of 1-150 such that the sum doesn't exceed of~250.

schedule_change does not set SetIdSession and accordingly would cause any usage of it to not be able to process equivocations, AFAIUI.
